### DISCUSSION TITLE
What's the way to get advanced parameter set by the json_file_path?

### DISCUSSION THREAD

**Discussion by alonsnir** [realsenseai/realsense-ros#2812] [Q&A] [answered]
So I set one of the advanced parameters via the json_file_path. How do I get it from the running node?

Sample invoke:
ros2 run realsense2_camera realsense2_camera_node --ros-args -p rgb_camera.profile:=1280x720x15 -p depth_module.profile:=1280x720x15 -p align_depth.enable:=true -p json_file_path:=/home/host1/camera-settings.json

I'm looking for a way to query one of the parameters from that camera-settings.json

**Comment by MartyG-RealSense** [comment]
Hi @alonsnir  You may be able to query the current value of a json-defined Advanced Mode parameter from ROS by writing a Python **node script** like the examples provided by the RealSense ROS2 wrapper at the link below.

https://github.com/IntelRealSense/realsense-ros/tree/ros2-development/realsense2_camera/scripts

**Reply by alonsnir** [reply]
Hi @MartyG-RealSense I'm struggling to get it. Looking at the samples you have referred to, I find subs for available topics e.g. /camera/depth/image_rect_raw, etc. I do not find any of the "advanced" parameters, could you please elaborate a bit of how to get them?

**Comment by MartyG-RealSense** [comment, marked as answer]
A ROS node script provides a way to interface with RealSense SDK (librealsense) settings that may not be accessible directly from the ROS wrapper.  I do not know of a node script example that demonstrates retrieving Advanced Mode parameters from the ROS wrapper though, unfortunately.

The Python wrapper of librealsense has an Advanced Mode example script that you could use as a foundation for working out the code for a ROS node script.  I do not have a finished Advanced Mode node script for ROS to provide though.

https://github.com/IntelRealSense/librealsense/blob/master/wrappers/python/examples/python-rs400-advanced-mode-example.py
